src/main/java/uk/gov/ons/ctp/response/action/scheduled/distribution/ActionDistributor.java

```diff
@@ -51,74 +56,37 @@ public ActionDistributor(
   /**
    * Called on schedule to check for submitted actions then creates and distributes the actions to
    * action exporter or notify gateway
-   *
-   * @return the info for the health endpoint regarding the distribution just performed
    */
-  @Transactional
-  public DistributionInfo distribute() {
-    final DistributionInfo distInfo = new DistributionInfo();
+  public void distribute() {
     List<ActionType> actionTypes = actionTypeRepo.findAll();
-
-    for (final ActionType actionType : actionTypes) {
-      final List<InstructionCount> instructionCounts = processActionType(actionType);
-      distInfo.getInstructionCounts().addAll(instructionCounts);
-    }
-    return distInfo;
+    actionTypes.forEach(this::processActionType);
   }
 
-  private List<InstructionCount> processActionType(final ActionType actionType) {
-    final InstructionCount requestCount =
-        InstructionCount.builder()
-            .actionTypeName(actionType.getName())
-            .instruction(DistributionInfo.Instruction.REQUEST)
-            .count(0)
-            .build();
-    final InstructionCount cancelCount =
-        InstructionCount.builder()
-            .actionTypeName(actionType.getName())
-            .instruction(DistributionInfo.Instruction.CANCEL_REQUEST)
-            .count(0)
-            .build();
-
+  private void processActionType(final ActionType actionType) {
+    String actionTypeName = actionType.getName();
+    RLock lock = redissonClient.getFairLock(lockPrefix + actionTypeName);
```

Comment:
you need to lock the lock after you get it

Reply:
🤦‍♂️ yup that would make sense